File: bindings/py/python/sysand/_include.py
# SPDX-License-Identifier: MIT OR Apache-2.0
# SPDX-FileCopyrightText: © 2025 Sysand contributors <opensource@sensmetry.com>

from __future__ import annotations

from . import _sysand_core as sysand_rs

from pathlib import Path
from typing import Literal


def include(
    *,
    path: str | Path,
    src_path: str | Path,
    compute_checksum: bool = False,
    index_symbols: bool = True,
    force_format: Literal["sysml", "kerml"] | None = None,
) -> None:
    sysand_rs.do_include_py(
        str(path), str(src_path), compute_checksum, index_symbols, force_format
    )


__all__ = ["include"]

File: sysand/src/commands/include.rs
// SPDX-License-Identifier: MIT OR Apache-2.0
// SPDX-FileCopyrightText: © 2025 Sysand contributors <opensource@sensmetry.com>

use anyhow::{Result, bail};
use camino::Utf8PathBuf;
use sysand_core::{context::ProjectContext, include::do_include, project::utils::wrapfs};

use crate::CliError;

pub fn command_include(
    paths: Vec<Utf8PathBuf>,
    compute_checksum: bool,
    index_symbols: bool,
    ctx: ProjectContext,
) -> Result<()> {
    let mut current_project = ctx
        .current_project
        .ok_or(CliError::MissingProjectCurrentDir)?;

    let mut unix_paths = Vec::with_capacity(paths.len());
    for p in paths {
        if !wrapfs::is_file(&p)? {
            bail!("`{p}` does not exist or is not a file");
        }
        unix_paths.push(current_project.get_unix_path(p)?);
    }
    do_include(
        &mut current_project,
        unix_paths.into_iter(),
        compute_checksum,
        index_symbols,
        None,
    )?;

    Ok(())
}
